guide/interactions/registering-slash-commands.md

```diff
@@ -156,24 +293,208 @@ const data = new SlashCommandBuilder()
-		subcommand
-			.setName('user')
-			.setDescription('Info about a user')
-			.addUserOption(option => option.setName('target').setDescription('The user')))
-	.addSubcommand(subcommand =>
-		subcommand
-			.setName('server')
-			.setDescription('Info about the server'));
+For those developers looking to make more organized and complex groups of commands, look no further than subcommands and groups.
+
+**Subcommands** organize your commands by **specifying actions within a command or group**.
+
+**Subcommand Groups** organize your **subcommands** by **grouping subcommands by similar action or resource within a command**.
+
+These are not enforced rules. You are free to use subcommands and groups however you'd like.
+
+::: warning
+Currently, subcommands and subcommand groups all appear at the top level in the command explorer. This may change in the future to include them as nested autocomplete options.
+:::
+
+::: danger
+Using subcommands or subcommand groups will make your base command unusable. You can't send the base `/permissions` command as a valid command if you also have `/permissions add | remove` as subcommands or subcommand groups
+:::
+
+Discord supports nesting one level deep within a group, meaning your top level command can contain subcommand groups, and those groups can contain subcommands. **That is the only kind of nesting supported**. Here's some visual examples:
+
+```:no-line-numbers
+VALID
+
+command
+|
+|__ subcommand
+|
+|__ subcommand
+
+----
+
+command
+|
+|__ subcommand-group
+    |
+    |__ subcommand
+|
+|__ subcommand-group
+    |
+    |__ subcommand
+
+
+-------
+
+INVALID
+
+
+command
+|
+|__ subcommand-group
+    |
+    |__ subcommand-group
+|
+|__ subcommand-group
+    |
+    |__ subcommand-group
+
+----
+
+INVALID
+
+command
+|
+|__ subcommand
+    |
+    |__ subcommand-group
+|
+|__ subcommand
+    |
+    |__ subcommand-group
 ```
 
+**Example walkthrough**
+
+Let's look at an example. Let's imagine you run a moderation bot. You want to make a `/permissions` command that can do the following:
+
+- **Get** the guild permissions for a **user** or a **role**
+- **Get** the permissions for a **user** or a **role** on a specific channel
+- **Edit** the guild permissions for a **user** or a **role**
+- **Edit** the permissions for a **user** or a **role** on a specific channel
+
+We'll start by defining the top-level information for `/permissions`:
+
+![](./images/subcommand-example.png)
+
+```js:no-line-numbers
+new SlashCommandBuilder()
+  .setName('permissions')
+  .setDescription('Get or edit permissions for a user or a role')
+```
+
+Now we have a command named `permissions`. We want this command to be able to affect **users** and **roles**. Rather than making two separate commands (for example, `permissions-users` and `permissions-role`), we can use subcommand groups. We want to use subcommand groups here because we are grouping subcommands (`get` and `edit`) on a similar resource: `user` or `role`.
```

Comment:
```suggestion
Now you have a command named `permissions` and want this command to be able to affect **users** and **roles**. Rather than making two separate commands (for example, `permissions-users` and `permissions-role`), you can use subcommand groups. It's recomended to use subcommand groups here because you are grouping subcommands (`get` and `edit`) on a similar resource: `user` or `role`.
```